### ISSUE TITLE
OpenCV_4.5.1-foss-2020b-contrib sanity check error on libopencv_freetype.so

### ISSUE THREAD

**Issue by guimou** [easybuilders/easybuild-easyconfigs#12491]
Hi!
Trying to build OpenCV_4.5.1-foss-2020b-contrib, I get this error:
`Sanity check failed: no file found at 'lib64/libopencv_freetype.so' in /opt/apps/easybuild/software/OpenCV/4.5.1-foss-2020b-contrib`
I checked in the folder, and it seems that libopencv_freetype is indeed not built. I checked the log, but it does not seem to complain about anything, only the sanity check is failing.
Should it be removed from sanity check, or am I missing something?
I'm tagging you, @boegel because you're the one who commited this.
Thanks,
Guillaume.

**Comment by Micket** [comment]
We'll need to see the entire build log file to be able to say anything.

**Comment by Micket** [comment]
```
-- freetype2:   NO
```

during configure step. Checking my build
``
-- Checking for module 'freetype2'
--   Found freetype2, version 23.3.17
-- Checking for module 'harfbuzz'
--   Found harfbuzz, version 2.6.7
-- freetype2:   YES (ver 23.3.17)
```

but that version number doesn't make any sense...

**Comment by guimou** [comment]
You mean the vesion number from freetype2? This seems to be the freetype-config version. On my system (different than where I built the package):
```
freetype-config --version
23.4.17
```
So it seems coherent, no?

**Comment by guimou** [comment]
From the Freetype2 doc:

---
For each new public release of FreeType 2, there are generally *three* distinct `version' numbers to consider:
* The official FreeType 2 release number, like 2.7.0 or 2.10.2.
* The  libtool (and  Unix)  specific version  number, like  23.2.17.
This is what  pkg-config freetype2 --modversion or freetype-config --version returns.
* The platform-specific shared object  number, used for example when the library is installed as `/usr/lib/libfreetype.so.6.17.2'.
---
Ref: https://github.com/servo/libfreetype2/blob/master/freetype2/docs/VERSIONS.TXT

**Comment by Micket** [comment]
Well then, I have even less of an idea. At least we know it's broken at the configure step already, but not really why. The contrib stuff doesn't help either as makes it even messier.
I'm short on ideas, typically these issues stem from
1. Local easybuild changes to configuration/hook
2. Host machine having unexpected libraries installed and things gets picked up when they shouldn't
3. We have forgotten to add something like a pkg-config as a necessary builddep in the easyconfig, and just haven't noticed it because our build machines happened to have an OS-installed version that worked by accident.

**Comment by guimou** [comment]
Yes, all of this makes sense. And I'm still digging anyway, finding references here and there that may (or may not...) be related. Like :
> However, I do think this is still a problem still because the conda-forge version of opencv didn't check to see that freetype was of a recent enough vintage. I had originally installed the 'defaults' channel version of opencv (version 3.4.2) which uses an older freetype. I am suggesting that freetype>=2.10.0 be listed as a dependency, since older versions of freetype are not supported by opencv 4.1.0.
Ref: https://github.com/conda-forge/opencv-feedstock/issues/174
